src/server/game/Entities/Object/Object.h

```diff
@@ -487,6 +487,7 @@ class WorldObject : public Object, public WorldLocation
         bool IsWithinDistInMap(WorldObject const* obj, float dist2compare, bool is3D = true) const;
         bool IsWithinLOS(float x, float y, float z) const;
         bool IsWithinLOSInMap(WorldObject const* obj) const;
+        Position GetHitSpherePointFor(Position const& dest) const;
```

Comment:
Might want to consider providing an overload like
`void GetHitSpherePointFor(Position const& dest, float& x, float& y, float& z) const;`
as well for use cases (like your own) that want to work directly with coords instead of having to construct temporary position objects.